### ISSUE TITLE
Recently, I found some problems when I experimented with your code

### ISSUE THREAD

**Issue by xiaomao19970819** [jaeyoon1603/Retrieval-RegionalAttention#7]
I want to make sure that the parameters of resnet-101 pre-training in Imagenet are not fine-tuned again?
Because when I replaced your code Caffe model parameter with pytorch's official Resnet-101 parameter, the performance dropped dramatically.

**Comment by jaeyoon1603** [comment]
We trained the only attention model while freezing the feature backbone network. The performance drop may occur in a case where a new module(e.g., attention) is suddenly added to a pre-trained module.

**Comment by xiaomao19970819** [comment]
> We trained the only attention model while freezing the feature backbone network. The performance drop may occur in a case where a new module(e.g., attention) is suddenly added to a pre-trained module.

Thank you very much for your reply. I noticed that you read the images with Opencv, which is different from the way of pytorch pre-training model (PIL), which leads to the different channel order of the images.
This is the reason why my experiment did not work well

**Comment by xiaomao19970819** [comment]
> We trained the only attention model while freezing the feature backbone network. The performance drop may occur in a case where a new module(e.g., attention) is suddenly added to a pre-trained module.

What resolution ( 800x800 ? ) do you use to pre-train the Resnet101 backbone on Imagenet ? 

I used the pre-training Resnet101 model officially provided by Pytorch, which also fixed the feature extraction layer and only trained the attention module. However, after four epoches (each epoch goes through the whole Imagenet dataset), the test results on the Oxford dataset were poor

**Comment by jaeyoon1603** [comment]
I just used the off-the-shelf ResNet101 and it is pre-trained from Caffe, not from Pytorch.
You also need the weights of Caffe ResNet101 If you reproduce our result.
To do so, I recommend you to extract only ResNet101 weights, which was frozen during attention training, from our pre-trained model in https://sglab.kaist.ac.kr/RegionalAttention/weights.tar. 

Or if you want to build along with the Pytorch ResNet101 model, you may adjust the RGB value scale and RGB order of an input image since the two frameworks employ different input shapes, as far as I know.

**Comment by xiaomao19970819** [comment]
> I just used the off-the-shelf ResNet101 and it is pre-trained from Caffe, not from Pytorch.
> You also need the weights of Caffe ResNet101 If you reproduce our result.
> To do so, I recommend you to extract only ResNet101 weights, which was frozen during attention training, from our pre-trained model in https://sglab.kaist.ac.kr/RegionalAttention/weights.tar.
> 
> Or if you want to build along with the Pytorch ResNet101 model, you may adjust the RGB value scale and RGB order of an input image since the two frameworks employ different input shapes, as far as I know.

Thank you for your suggestion. I use PIL to read the pictures, so there will be no problem with the RGB channel order.
I'm curious how many times you iterate while training the attention module, and I notice that it's 999999 in the exposed code.
In addition, your paper said that the learning rate of the model would be dynamically adjusted, which I did not find in the code.
On the other hand, the PCA parameters I trained by myself are far less effective than those you provided. Could you please provide me with a training code, which will be very helpful for me to reproduce your work.
Thank you again

**Comment by jaeyoon1603** [comment]
Sorry for the late response. 
I actually lost the historical information (the number of iterations) of the whole training but I stopped the training based on a validation error as following in the paper. 
I missed reflecting two steps of the dynamic learning rate in the final code but you can simply adjust the learning rate based on validation error. 
As for PCA training, I used PCA decomposition module of scikit-learn (https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html) but you may transpose the PCA parameter matrix to use in Pytorch.

**Comment by xiaomao19970819** [comment]
> Sorry for the late response.
> I actually lost the historical information (the number of iterations) of the whole training but I stopped the training based on a validation error as following in the paper.
> I missed reflecting two steps of the dynamic learning rate in the final code but you can simply adjust the learning rate based on validation error.
> As for PCA training, I used PCA decomposition module of scikit-learn (https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html) but you may transpose the PCA parameter matrix to use in Pytorch.

I converted the resnet101 Caffe model provided by Kaiming He into pytorch format (RGB input format).
Using this model, I trained the attention module, and only 100  iterations could get 67% mAP on Oxford (without adding PCA layer).
However, when I switched to pytorch official Resnet101 model for the backbone network, after 100 iterations, there was only 51% of the Map.
It's very confusing to me. Why do the Caffe model work so much better than the Pytorch model
